client/src/ts/streaming_graph.ts

```diff
@@ -16,153 +16,80 @@ export class TimeWindow {
     return this.cached;
   }
 
-  // decorator to make getting the beginning nicer.
+  // Method to make getting the beginning nicer.
   public begin(): number { return this.cached[0]; }
 
-  // decorator to make getting the end nicer.
+  // Method to make getting the end nicer.
   public end(): number { return this.cached[1]; }
 }
 
-// adapter class to make a Plottable.Scale have dynamically updating domain.
-export class WindowedScale {
-  public domainUpdater: (domain: any[]) => any[];
-  private _scale: Plottable.Scale<any, any>;
+// Decorator class to make a Plottable.Scale have dynamically updating domain.
+export class WindowedScale extends Plottable.Scales.Linear {
+  public domainUpdater: (domain: any[]) => number[];
+  private _domain: number[];
 
-  constructor(scale: Plottable.Scale<any, any>,
-              domainUpdater: (domain: any[]) => any[]) {
+  constructor(domainUpdater: (domain: any[]) => any[]) {
+    super();
     this.domainUpdater = domainUpdater;
-    this._scale = scale;
   }
 
-  // returns a copy of the scale that is updated to the latest domain.
-  public getScale(): Plottable.Scale<any, any> {
-    if (this.domainUpdater) {
-      this._scale.domain(this.domainUpdater(this._scale.domain()));
+  // Updates the domain.
+  public domain(): number[];
+  public domain(domain: any[]): Plottable.QuantitativeScale<number>;
+  public domain(domain?: any[]): number[]|Plottable.QuantitativeScale<number> {
+    if (domain) {
+      return super.domain(domain);
+    } else if (this.domainUpdater) {
+      super.domain(this.domainUpdater(super.domain()));
     }
-    return this._scale;
+    return super.domain();
   }
 }
 
-// adapter class to make a Plottable.Dataset compatible with streaming data by
+// Decorator class to make a Plottable.Dataset compatible with streaming data by
 // providing concise windowing and add data functions.
-export class StreamingDataset {
+export class StreamingDataset extends Plottable.Dataset {
   public dataUpdate: () => void;
-  private _dataset: Plottable.Dataset;
   private _filter: (data: any[]) => any[];
 
   constructor(data: any[], metadata: any) {
-    this._dataset = new Plottable.Dataset(data, metadata);
+     super(data, metadata);
   }
 
-  // return or update the data owned by the dataset.
+  // Return or update the data owned by the dataset.
   public data(): any[];
   public data(data: any[]): this;
   public data(data?: any[]): this|any[] {
     if (data) {
-      this._dataset.data(data);
+      if (this._filter) {
```

Comment:
Do you think it would make sense to provide an identity function as default, so you can simplify this logic? Or is it not worth it?